velox/serializers/PrestoSerializer.cpp

```diff
@@ -1935,6 +2001,39 @@ void serializeFlatVector<TypeKind::BOOLEAN>(
   }
 }
 
+template <>
+void serializeFlatVector<TypeKind::OPAQUE>(
```

Comment:
This implementation is used by the `PrestoBatchVectorSerializer` only. You also need to add an implementation for `PrestoIterativeVectorSerializer`. I am not sure which one you use for your use-case. To give you more context the `PrestoBatchVectorSerializer` is optimized for serializing full vectors whereas `PrestoIterativeVectorSerializer` is used for serializing subsets of rows scattered within a single vector(which is usually used for a partitioning exchange).

the `serializeFlatVector` implementation for `PrestoIterativeVectorSerializer` is here: https://github.com/facebookincubator/velox/blob/a33e8d733d9ed8f23121ec2737a55c56a9f740a8/velox/serializers/PrestoSerializer.cpp#L2737

Finally, for a complete implementation, you also need to implement `estimateSerializedSize` for both types of serializers. This is used by the PartitionedOutput node to figure out when it needs to flush data to the wire.